src/librustc_mir/transform/mod.rs

```diff
@@ -62,7 +62,7 @@ fn mir_keys(tcx: TyCtxt<'_>, krate: CrateNum) -> &DefIdSet {
     let mut set = DefIdSet::default();
 
     // All body-owners have MIR associated with them.
-    set.extend(tcx.body_owners());
+    set.extend(tcx.body_owners().map(LocalDefId::to_def_id));
```

Comment:
This query should return a `&'tcx FxHashSet<LocalDefId>` instead of `&'tcx DefIdSet` (which AFAIK is just `FxHashSet<DefId>` - IMO we should stop using the aliases, they only exist because `NodeMap` was the first iteration of `FxHashMap`, ages ago, and we never moved on from that).

Reply:
(this doesn't have to be done in this PR, especially if you plan to tweak queries in a different PR)